### PR TITLE
Bug 1949799: Set maximum for ingresscontroller spec.tuningOptions.threadCount to 64

### PR DESCRIPTION
**Pull request by rfredette**
As of this PR, HAProxy only supports up to 64 threads

### CODE REVIEW

File: operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml [hunk 1]
```diff
@@ -990,6 +990,7 @@ spec:
                       Reducing the number of threads may cause the ingress controller
                       to perform poorly."
                     format: int32
+                    maximum: 64
```
Comment (frobware) [inline operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml:993]: Maybe we should qualify why 64 is a max (i.e., it's not arbitrary, but imposed by haproxy-2.2.x).

Comment (rfredette) [reply operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml:993]: That's a good idea. I'll update the description.

File: operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml [hunk 2]
```diff
@@ -981,15 +981,17 @@ spec:
...
+                      64 threads. If this field is empty, the IngressController will
+                      use the default value.  The current default is 4 threads, but
+                      this may change in future releases. \n Setting this field is
+                      generally not recommended. Increasing the number of HAProxy
                       threads allows ingress controller pods to utilize more CPU time
                       under load, potentially starving other pods if set too high.
                       Reducing the number of threads may cause the ingress controller
                       to perform poorly."
                     format: int32
+                    maximum: 64
```
Comment (soltysh) [inline operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml:994]: You're changing a value from previously unbounded? What is the chance that this value is currently set higher than 64 in existing clusters?

Comment (rfredette) [reply operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml:994]: The field is new in 4.8, but even if someone is running a 4.8 pre-release build and `threadCound` is set above 64, HAProxy exits with a fatal error and the router pods quickly end up in CrashLoopBackOff state, so it shouldn't be possible that a working cluster has `threadCount` set above 64.

**Comment by frobware** [comment]
I think the review comments are addressed. Can we move this along now?

**Pull request merged**